src/components/utils/src/threads/thread_posix.cc

```diff
@@ -261,15 +270,27 @@ void Thread::stop() {
   SDL_DEBUG("Stopped thread #" << handle_ << " \"" << name_ << " \"");
 }
 
-void Thread::join() {
+void Thread::join(const bool force_stop) {
   SDL_AUTO_TRACE();
   DCHECK_OR_RETURN_VOID(!IsCurrentThread());
 
-  stop();
+  if (force_stop) {
+    stop();
+  }
 
```

Comment:
@anosach-luxoft, I agree (actually I mentioned it in one of my emails). I can not use recursive mutex here. If I can - I'll put locking at the beginning without worrying for double lock in stop(). I can not because state_lock_ is used with condvars. It is not allowed then to be recursive because of condvar compatibility with Qt and so on.
Any (acceptable) suggestions?